content/client-lib-development-guide/features.textile

```diff
@@ -367,6 +367,9 @@ h3(#realtime-connection). Connection
 ** @(RTN19a)@ Any @ProtocolMessage@ that is awaiting an @ACK@/@NACK@ on the old transport will not receive the @ACK@/@NACK@ on the new transport. The client library must therefore resend any @ProtocolMessage@ that is awaiting a @ACK@/@NACK@ to Ably in order to receive the expected @ACK@/@NACK@ for that message. The Ably service is responsible for keeping track of messages, ignoring duplicates and responding with suitable @ACK@/@NACK@ messages
 ** @(RTN19b)@ If there are any pending channels i.e. in the @ATTACHING@ or @DETACHING@ state, the respective @ATTACH@ or @DETACH@ message should be resent to Ably
 ** @(RTN19b)@ If a @SYNC@ is underway, ensure the client library adheres to @RTP3@
+* @(RTN20)@ Heartbeats
+** @(RTN20a)@ If a transport does not receive a heartbeat for a period greater than the sum of the @heartbeatPeriod@ (sent in the @connectionDetails@ in the @CONNECTED@ message received on that transport) and the @realtimeRequestTimeout@, that transport may be disconnected. This requirement is optional; in deciding whether to implement, client library developers should take into account whether the transport in question is susceptible to undetected dropped connections.
```

Comment:
@SimonWoolf I think it's too tight because if for example a frontend was under extreme load and was delayed in sending the Heartbeat + the user was on a bad connection, we'd just end up creating more work for that frontend by terminating the connection and immediately resuming a new one.  Why not state that once `realtimeRequestTimeout` has failed, the client initiates a Heartbeat and if a response is not received within `realtimeRequestTimeout` then it disconnects the connection.  I realise we would prefer that users get feedback more quickly when a connection drops, but TCP/IP and/or the OS should do that in most cases, and where it doesn't, I think we'll have accept that it's a bit harder for us to be sure the connection is indeed gone.

BTW. Why did you not include an option to include an ID in Heartbeats so that clients can correlate the Heartbeat they sent with the one they received?  Seems like a trivial change.